Dynatrace/dynatrace-operator
# Move hash generation and adding to generic query implementation

## Description

https://dt-rnd.atlassian.net/browse/DAQ-1391

With this the generation and adding of the `template-hash` annotation for objects is added to the generic query implementation. The hash is always generated and set when the object is created or updated.

**NOTE:** I removed the previous way of setting the annotation where it is not needed anymore because I moved it into the query implementation. If you find places where I missed something to remove please let me know.



## How can this be tested?

Deploy the operator and have a look if everything runs, you can also check if the `kspm` secret for example has a `template-hash` annotation set. Other example you might want to check: 

- edgeconnect secret
- logmonitoring config secret
- activeGate auth token
- ...
